README.md

```diff
@@ -24,88 +22,117 @@ Via CDN:
 
- Note that msal.js is built for ES5, therefore enabling support for Internet Explorer 11. If you want to target Internet Explorer, you'll need to add a reference to promises polyfill.
+Internet Explorer does not have native `Promise` support, and so you will need to include a polyfill for promises such as `bluebird`.
 
     <!-- IE support: add promises polyfill before msal.js  -->
     <script src="https://cdnjs.cloudflare.com/ajax/libs/bluebird/3.3.4/bluebird.min.js" class="pre"></script>
 
-See here for more details on [supported browsers](https://github.com/AzureAD/microsoft-authentication-library-for-js/wiki/FAQs#q4-what-browsers-is-msaljs-supported-on).
+See here for more details on [supported browsers and known compatability issues](https://github.com/AzureAD/microsoft-authentication-library-for-js/wiki/FAQs#q4-what-browsers-is-msaljs-supported-on).
+
+## Roadmap and What To Expect From This Library
+Msal support on Javascript is a collection of libraries. msal-core or just simply msal, is the framework agnostic we library. Once our 1.X+ is stabilized, we are going to bring our msal-angular library to 1.x parity.  We are currently no planning on supporting msal-angularjs for 1.x, because the feedback we have heard is that community has moved on, however we will provide some code patterns of how you can use msal-core in an angularjs app. After our current libraries are up to standards, we will begin balancing new feature requests, with new platforms such as `react`, server side `node.js`, `electron`, etc...
+
+Our goal is communicate extremely well with the community and to take their opinions into account. We would like to get to a monthly minor release schedule, with patches comming as often as needed.  The level of communication, planning, and granularity we want to get to will be a work in progress.
+
+Please check our [roadmap](https://github.com/AzureAD/microsoft-authentication-library-for-js/wiki#roadmap) to see what we are workign on and what we are tracking next.
+
+## Oauth and the Implict Flow
+Msal implments the [Implicit Grant Flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-implicit-grant-flow), as defined by the Oauth 2.0 protocol and is [OpenID](https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-protocols-oidc) compliant.
+
+Our goal is that the library abstracts enough of this away that you can get plug and play authentication, but it is important to know for users that need AAD functionality outside of simple auth, and also important to understand from a security perspective.  There is no client secret, and one less hop between client and server, and so tokens are naturally less secure.  This is mitigated by the use of tokens with a short time to live. We encourage you to read about the flows, and the spec, so that you have a strong understanding of the use case.
+
+## Cache Storage
```

Comment:
I suggest we move the topics cache storage, oauth 2.0 implicit details, Roadmap sections after the Usage section. 
It is common in most Readmes to expect to see a Installation followed by Usage for those who want to get quickly started.

Reply:
Can we add hyperlinks to these and add more data as needed?

Reply:
@navyasric  I thought about that, but some of the feedback we heard was that we should be more upfront with some of this information.  If you still think below usage is better I am find with that.

Reply:
@sameerag  do we have links for cache ?

Reply:
no, we should document what we cache and where. I am not sure if there is a specific section except the 'localStorage' vs 'sessionStorage' setting. @navyasric can correct me if I an wrong.

Reply:
No section describing how and what we cache. Only info on how to configure cache. Do we really think developers should know internal details of how library caches? Probably an advanced wiki topic than on the Readme page.

Reply:
Yes having these info on Readme is being more visible and upfront. But I still think installation and usage should be near each other and the first things most developers want to see to get quickly started. Those who need further info will read further.